### PR TITLE
🛠 Wiki link, cross-ref errors, and docs updates

### PR DESCRIPTION
**Pull request by rowanc1**

### CODE REVIEW

**Comment by rowanc1** [comment]
@fwkoch this addresses quite a few of the issues we found with cross-referencing on Friday.

File: docs/cross-references.md [hunk 1]
```diff
@@ -128,7 +124,7 @@ These will show up, for example, as `Section 1` and `Section 2.1`.
 
 (heading-numbering)=
 
-## Numbering
+## Heading Numbering
```
Comment (fwkoch) [inline docs/cross-references.md]:
```suggestion
## Header Numbering
```
for consistency? probably doesn't matter

File: packages/myst-transforms/src/enumerate.ts [hunk 2]
```diff
@@ -327,7 +330,7 @@ export class ReferenceState implements IReferenceState {
         copyNode(target.node as Heading).children as PhrasingContent[],
       );
     } else if (target.kind === TargetKind.equation) {
-      fillReferenceEnumerators(this.file, node, '(%s)', target.node.enumerator, 'Equation');
```
Comment (fwkoch) [inline packages/myst-transforms/src/enumerate.ts:330]: Hah, easy fix...

**Review by fwkoch — APPROVED** [review]
All looks straightforward - docs, errors, wiki links...

File: packages/myst-transforms/src/enumerate.ts (hunk 2 above)
Comment (rowanc1) [reply packages/myst-transforms/src/enumerate.ts:330]: Nice when that happens. :)

**Pull request merged**